### PR TITLE
fix: create auto ml job for tests that based on existing job

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*

*Description of changes:*
For auto ml jobs, create a base auto ml job for tests that base on an existing auto ml job.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integ/test_auto_ml.py [hunk 1]
```diff
@@ -62,6 +62,11 @@
 }
 
 
+@pytest.fixture(scope="module", autouse=True)
+def create_auto_ml_job_if_not_exist(sagemaker_session):
```
Comment (ChoiByungWook) [inline tests/integ/test_auto_ml.py]: what test is using this fixture?

Comment (ChuyangDeng) [reply tests/integ/test_auto_ml.py]: This fixture is autouse. But now a second thought made me realized not all tests in the class are using this fixture, I will apply it to individual tests.

File: tests/integ/auto_ml_utils.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    auto_ml_job_name = "python-sdk-integ-test-base-job"
+    auto_ml = AutoML(
+        role=ROLE,
+        target_attribute_name=TARGET_ATTRIBUTE_NAME,
+        sagemaker_session=sagemaker_session,
+        max_candidates=3,
+    )
+
+    try:
+        auto_ml.describe_auto_ml_job(job_name=auto_ml_job_name)
```
Comment (knakad) [inline tests/integ/auto_ml_utils.py]: Don't fix it as part of this PR, but we should have a session method for this if the base object isn't needed. Feel free to chat in person if I'm missing something.

i.e.:
session.describe_auto_ml_job(job_name)

Meaning that, if we don't need anything from the object, we should just use a session method.

Comment (ChuyangDeng) [reply tests/integ/auto_ml_utils.py]: Got it! Thanks!

**Pull request merged**